### PR TITLE
Interim commit

### PR DESCRIPTION
**Pull request by larubbio**
This adds database constraints and some fields that I think are required.  It is not the complete fix for #99.  I'm still waiting for final word on the location requirements.  I'm also waiting for a fix to the detached field issue.  I would like to merge this so the work doesn't get stale.

### CODE REVIEW

File: location/src/main/java/org/motechproject/nms/location/domain/HealthBlock.java [hunk 1]
```diff
@@ -19,18 +19,18 @@
 public class HealthBlock extends MdsEntity {
 
     @Field
-    @UIDisplayable(position = 0)
+    @Column(allowsNull = "false")
```
Comment (frankhuster) [inline location/src/main/java/org/motechproject/nms/location/domain/HealthBlock.java]: What's the difference between `@Column(allowsNull = "false")` and `@NotNull`?

**Comment by frankhuster** [comment]
Oh, and why remove the UI ordering?

**Comment by frankhuster** [comment]
lgtm

File: location/src/main/java/org/motechproject/nms/location/domain/HealthBlock.java (hunk 1 above)
Comment (larubbio) [inline location/src/main/java/org/motechproject/nms/location/domain/HealthBlock.java]: @Column(allowsNull = "false") adds a not null constraint to the database.  @NotNull does a validation in code and throws an exception before saving if the value is null.  So one is in code and the other is in the database.  (If I had to choose only one I would go with @Column

**Comment by larubbio** [comment]
I figured we could add the UI ordering if there is a story/need for it.

**Pull request merged**